pkg/oc/cli/create/route.go

```diff
@@ -77,11 +77,12 @@ func (o *CreateRouteSubcommandOptions) Complete(f kcmdutil.Factory, cmd *cobra.C
 		return err
 	}
 
-	o.KubeClient, err = f.ClientSet()
+	clientConfig, err := f.ToRESTConfig()
 	if err != nil {
 		return err
 	}
-	clientConfig, err := f.ToRESTConfig()
+
+	o.KubeClient, err = kexternalclientset.NewForConfig(clientConfig)
```

Comment:
How about using more narrow client? I'm shooting core (without looking at code), which should suffice here.